rfd/00xx-kubernetes-access-cluster-discovery.md

```diff
@@ -0,0 +1,362 @@
+
+The [AWS IAM authenticator][awsiamauthenticator] database is a simple `configmap/aws-auth` stored in `kube-system` namespace. This configmap cannot be changed unless Teleport has write permissions to Kubernetes Configmaps, which can only happen if the Teleport role is already in the configmap.
+
+ The configmap has the following format:
+
+```yaml
+apiVersion: v1
+data:
+  mapRoles: |
+    - groups:
+      - {kube group}
+      rolearn: {IAM role}
+      username: {user name}
+  mapUsers: |
+    - groups:
+      - {kube group}
+      userARN: {IAM user}
+      username: {user name}
+...
+```
+
+For Teleport to be able to connect and operate the Kubernetes cluster, it is required that the `configmap/aws-auth` maps the Teleport role into a Kubernetes group.
+
+```yaml
+apiVersion: v1
+data:
+  mapRoles: |
+    - groups:
+      - system:masters  # to be defined depending on the instalation method #
+      rolearn: arn:aws:iam::222222222222:role/teleport-role
+      username: system:teleport
+...
+```
+
+This means that this mapping has to exist for each cluster that the operator wants to be discovered and to do that, the operator has to change the `configmap/aws-auth` manually. `eksctl` has a simpler method for appending identities into the `configmap/aws-auth`, but it still requires executing it for every cluster that is expected to be enrolled by Teleport.
+
+```bash
+$  eksctl create iamidentitymapping --cluster  <clusterName> --region=<region> \
+      --arn arn:aws:iam::222222222222:role/teleport-role --group system:teleport --username system:teleport
+```
+
+When a user creates an Amazon EKS cluster, the AWS Identity and Access Management (IAM) entity user or role, such as a federated user that creates the cluster, is automatically granted `system:masters` permissions in the cluster's role-based access control (RBAC) configuration. If Teleport can use the same role as the user that created the cluster, then it is not required to change the `configmap`, but this solution requires that every cluster is created by users with the same IAM Role and has several security implications.
+
+### UX
+
+The Teleport configuration for automatic AWS EKS discovery will have the following structure:
+
+```yaml
+kubernetes_service:
+  enabled: yes
+  aws:
+  - regions: ["us-west-1"]
+    tags:
+      "env": "prod"
+
+    accounts:
+    - aws_account: "222222222222"
+      aws_roles:
+      - "arn:aws:iam::222222222222:role/teleport-role"
+
+  - regions: ["us-east-1", "us-east-2"]
+    tags:
+      "env": "stage"
+    aws_roles:
+    - "arn:aws:iam::222222222222:role/teleport-role"
+
+```
+
+To identify auto-discovered clusters, Teleport config will include a static label indicating it was created via auto-discovery.
+
+```json
+{
+    "kind": "kube_service",
```

Comment:
IIRC right now Kube clusters are stored as a part of the `ServerV2` resource. I think we should take this work as an opportunity to factor it out into a separate KubeService/KubeCluster resources like we did for databases and apps before: https://github.com/gravitational/teleport/pull/8140. ServerV2 was never meant to represent anything other than an SSH server.